processor/src/main/java/org/mapstruct/ap/processor/MethodRetrievalProcessor.java

```diff
@@ -80,19 +93,60 @@ public int getPriority() {
         return 1;
     }
 
+    private List<SourceMethod> retrievePrototypeMethods(TypeMirror typeMirror, MapperConfig mapperConfig) {
+        if ( typeMirror == null || typeMirror.getKind() == TypeKind.VOID ) {
+            return Collections.emptyList();
+        }
+
+        TypeElement typeElement = asTypeElement( typeMirror );
+        List<SourceMethod> methods = new ArrayList<SourceMethod>();
+        for ( ExecutableElement executable : getAllEnclosedExecutableElements( elementUtils, typeElement ) ) {
+
+            ExecutableType methodType = typeFactory.getMethodType( typeElement, executable );
+            List<Parameter> parameters = typeFactory.getParameters( methodType, executable );
+            boolean containsTargetTypeParameter = SourceMethod.containsTargetTypeParameter( parameters );
+
+            // prototype methods don't have prototypes themselves
```

Comment:
> Defining methods on config classes should be a rare cases.

Actually, I don't agree here. I think that this will be a prime use case for me.

> It's only needed if a) the return type is an abstract class or b) you don't want that method be usable as mapping method.

and c) In the very near future, The method to which it applies is a forged method: MapMapping and IterableMapping (as existing today) and PropertyMapping ( #60 ) as proposed by Andreas. :smile:. 

I also don't understand why the return type per-se should be abstract. A lot of JAXB generated code produces non-abstract classes that serve as extension base all over the place (and don't have the attribute 'abstract="true"'. Do we exclude those from this methodology?